### ISSUE TITLE
[FEATURE] Add option to return float in diffInMonths

### ISSUE THREAD

**Issue by kanatkubash** [briannesbitt/Carbon#1448]
Hello,

Currently Carbon returns integer value when invoking `diffInMonths`, 
Is it possible to return float value?

This could make possible checks of kind `more than n month`, `less than n month` . 
For example:
```php
$startDate=Carbon::parse('2018-01-01');
$endDate=Carbon::parse('2018-02-05');
/// Third argument indicates whether to return with remainder (float) or as integer
echo $startDate->diffInMonths($endDate,true,true);
/// ~ 1.13
```

**Comment by kylekatarnls** [comment]
Hi,

It's interesting. Be known for check more/less than n month, you can use:
```php
// start-end difference is more than n months
if ($endDate->subMonths($n) > $startDate) {}
// start-end difference is less than n months
if ($endDate->subMonths($n) < $startDate) {}
```

**Comment by kylekatarnls** [comment]
Hi, #1467 will implement **floatDiffIn{Unit}** methods, **{Unit}** can be basic units: Seconds/Minutes/Hours/Days/Months/Years or any of this unit with the prefix "Real" to calculate difference using timestamps (can have a different output with DST changes).

Note: with your example:
```php
Carbon::parse('2018-01-01')->floatDiffInMonths('2018-02-05'); // 1.1428571428571
```
Here is how I decided to calculate it:
2018-01-01:00:00:00 -> 2018-02-01 00:00:00 = 1 month
2018-02-01 00:00:00 -> 2018-02-05 00:00:00 = 4 days
2018-02 is 28 days long, so 4 days of this month is 4/28 month = 0.1428571428571 month

**Comment by kylekatarnls** [comment]
Released in 2.4.0
